--- a/quectel-qrtr-proxy.c
+++ b/quectel-qrtr-proxy.c
@@ -555,7 +555,7 @@ static int recv_qmi_from_client(PQCQMIMS
         }
         else if (le16toh(pQMI->CTLMsg.QMICTLMsgHdr.QMICTLType) == QMICTL_GET_CLIENT_ID_REQ) {
             uint8_t QMIType = pQMI->CTLMsg.GetClientIdReq.QMIType;
-            PQCQMIMSG pRsp = (PQCQMIMSG)malloc(256);
+            PQCQMIMSG pRsp = (PQCQMIMSG)malloc(sizeof(*pRsp));
 
             if (pRsp) {
                 uint8_t ClientId = 0;
@@ -589,7 +589,7 @@ static int recv_qmi_from_client(PQCQMIMS
             }
         }
         else if (le16toh(pQMI->CTLMsg.QMICTLMsgHdr.QMICTLType) == QMICTL_RELEASE_CLIENT_ID_REQ) {
-            PQCQMIMSG pRsp = (PQCQMIMSG)malloc(256);
+            PQCQMIMSG pRsp = (PQCQMIMSG)malloc(sizeof(*pRsp));
             release_qrtr_client_id(qrtr_con, pQMI->CTLMsg.ReleaseClientIdReq.QMIType, pQMI->CTLMsg.ReleaseClientIdReq.ClientId);
 
             if (pRsp) {
@@ -618,7 +618,7 @@ static int recv_qmi_from_client(PQCQMIMS
             }
         }
         else if (le16toh(pQMI->CTLMsg.QMICTLMsgHdr.QMICTLType) == QMICTL_GET_VERSION_REQ) {
-            PQCQMIMSG pRsp = (PQCQMIMSG)malloc(256);
+            PQCQMIMSG pRsp = (PQCQMIMSG)malloc(sizeof(*pRsp));
 
             if (pRsp) {
                 pRsp->QMIHdr.IFType = USB_CTL_MSG_TYPE_QMI;
