--- a/device.c
+++ b/device.c
@@ -342,16 +342,20 @@ BOOL qmidevice_detect(char *qmichannel,
             if (profile->usb_dev.idVendor == 0x2c7c) { //Quectel
                 switch (profile->usb_dev.idProduct) { //EC200U
                 case 0x0901: //EC200U
+                case 0x0902: //EC200D-EU
+                case 0x0904: //EC200G
                 case 0x8101: //RG801H
                     atIntf = 2;
                 break;
                 case 0x0900: //RG500U
+                case 0x0903: //EC600
                     atIntf = 4;
                 break;
                 case 0x6026: //EC200T
                 case 0x6005: //EC200A
                 case 0x6002: //EC200S
                 case 0x6001: //EC100Y
+                case 0x6007: //EG915Q-NA
                     atIntf = 3;
                 break;
                 default:
