From 391f406596e67d935746f93a163afec660f4c622 Mon Sep 17 00:00:00 2001
From: c0re100 <corehusky@gmail.com>
Date: Sat, 17 May 2025 22:14:08 +0800
Subject: [PATCH] Remove unused pref here too

---
 src/gui/optionsdialog.cpp | 1 -
 1 file changed, 1 deletion(-)

--- a/src/gui/optionsdialog.cpp
+++ b/src/gui/optionsdialog.cpp
@@ -1203,7 +1203,6 @@ void OptionsDialog::loadBittorrentTabOpt
 void OptionsDialog::saveBittorrentTabOptions() const
 {
     auto *session = BitTorrent::Session::instance();
-    auto *pref = Preferences::instance();
 
     session->setDHTEnabled(isDHTEnabled());
     session->setPeXEnabled(m_ui->checkPeX->isChecked());
