diff --git a/kernel/pf_ring.c b/kernel/pf_ring.c
index 702aa0a..22ce4b0 100644
--- a/kernel/pf_ring.c
+++ b/kernel/pf_ring.c
@@ -4916,7 +4916,12 @@ static void set_netdev_promisc(struct net_device *netdev) {
   if_flags = (short) dev_get_flags(netdev);
   if(!(if_flags & IFF_PROMISC)) {
     if_flags |= IFF_PROMISC;
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 4, 24)
+    dev_change_flags(netdev, if_flags, NULL);
+#else
     dev_change_flags(netdev, if_flags);
+#endif
+
   }
 
   rtnl_unlock();
@@ -4934,7 +4939,11 @@ static void unset_netdev_promisc(struct net_device *netdev) {
   if_flags = (short) dev_get_flags(netdev);
   if(if_flags & IFF_PROMISC) {
     if_flags &= ~IFF_PROMISC;
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 4, 24)
+    dev_change_flags(netdev, if_flags, NULL);
+#else
     dev_change_flags(netdev, if_flags);
+#endif
   }
 
   rtnl_unlock();
