--- a/configure.cmake
+++ b/configure.cmake
@@ -6,7 +6,7 @@ include(${CMAKE_CURRENT_LIST_DIR}/src/qd
 
 #### Tests
 
-qt_find_package(WrapLibClang 8 PROVIDED_TARGETS WrapLibClang::WrapLibClang)
+# qt_find_package(WrapLibClang 8 PROVIDED_TARGETS WrapLibClang::WrapLibClang)
 qt_find_package_extend_sbom(TARGETS WrapLibClang::WrapLibClang
     PACKAGE_VERSION "${QT_LIB_CLANG_VERSION}"
     LICENSE_EXPRESSION "Apache-2.0 WITH LLVM-exception"
