--- a/src/generator/config/nodemanip.h
+++ b/src/generator/config/nodemanip.h
@@ -1,6 +1,7 @@
 #ifndef NODEMANIP_H_INCLUDED
 #define NODEMANIP_H_INCLUDED
 
+#include <limits.h>
 #include <string>
 #include <vector>
 #include <quickjspp.hpp>
--- a/src/utils/file.h
+++ b/src/utils/file.h
@@ -1,6 +1,7 @@
 #ifndef FILE_H_INCLUDED
 #define FILE_H_INCLUDED
 
+#include <string.h>
 #include <string>
 
 #ifdef _WIN32
