diff --git a/config.c b/config.c
index 3ccad84..6b3a380 100644
--- a/config.c
+++ b/config.c
@@ -6,6 +6,7 @@
  *  published by the Free Software Foundation.
  */
 
+#include <linux/version.h>
 #include <linux/uaccess.h>
 #include <linux/inetdevice.h>
 #include <linux/seq_file.h>
@@ -20,6 +21,9 @@ static int update_gw_interface(const char *interface)
     int ret = 0;
     struct net_device *dev;
     struct in_device *in_dev;
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 3, 0)
+    struct in_ifaddr *ifa;
+#endif
 
     dev = dev_get_by_name(&init_net, interface);
     if (!dev) {
@@ -36,14 +40,22 @@ static int update_gw_interface(const char *interface)
         goto QUIT;
     }
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 3, 0)
+    in_dev_for_each_ifa_rcu(ifa, in_dev) {
+#else
     for_primary_ifa(in_dev) {
+#endif
         conf.interface_ipaddr = ifa->ifa_local;
         conf.interface_mask = ifa->ifa_mask;
         conf.interface_broadcast = ifa->ifa_broadcast;
 
         pr_info("Found ip from %s: %pI4\n", interface, &conf.interface_ipaddr);
         break;
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 3, 0)
+    }
+#else
     } endfor_ifa(in_dev)
+#endif
     
 QUIT:   
     dev_put(dev);
