--- a/qmi_wwan_q.c
+++ b/qmi_wwan_q.c
@@ -846,12 +846,20 @@ static struct rtnl_link_stats64 *_rmnet_
 		stats64 = per_cpu_ptr(dev->stats64, cpu);
 
 		do {
+#if (LINUX_VERSION_CODE < KERNEL_VERSION( 6,6,0 ))
 			start = u64_stats_fetch_begin_irq(&stats64->syncp);
+#else
+			start = u64_stats_fetch_begin(&stats64->syncp);
+#endif
 			rx_packets = stats64->rx_packets;
 			rx_bytes = stats64->rx_bytes;
 			tx_packets = stats64->tx_packets;
 			tx_bytes = stats64->tx_bytes;
+#if (LINUX_VERSION_CODE < KERNEL_VERSION( 6,6,0 ))
 		} while (u64_stats_fetch_retry_irq(&stats64->syncp, start));
+#else
+		} while (u64_stats_fetch_retry(&stats64->syncp, start));
+#endif
 
         stats->rx_packets += u64_stats_read(&rx_packets);
 		stats->rx_bytes += u64_stats_read(&rx_bytes);
