--- a/component/profile/cachefile/cache.go
+++ b/component/profile/cachefile/cache.go
@@ -13,7 +13,9 @@ import (
 )
 
 var (
-	fileMode os.FileMode = 0o666
+	initOnce     sync.Once
+	fileMode     os.FileMode = 0o666
+	defaultCache *CacheFile
 
 	bucketSelected = []byte("selected")
 	bucketFakeip   = []byte("fakeip")
@@ -134,8 +136,7 @@ func (c *CacheFile) Close() error {
 	return c.DB.Close()
 }
 
-// Cache return singleton of CacheFile
-var Cache = sync.OnceValue(func() *CacheFile {
+func initCache() {
 	options := bbolt.Options{Timeout: time.Second}
 	db, err := bbolt.Open(C.Path.Cache(), fileMode, &options)
 	switch err {
@@ -151,7 +152,14 @@ var Cache = sync.OnceValue(func() *Cache
 		log.Warnln("[CacheFile] can't open cache file: %s", err.Error())
 	}
 
-	return &CacheFile{
+	defaultCache = &CacheFile{
 		DB: db,
 	}
-})
+}
+
+// Cache return singleton of CacheFile
+func Cache() *CacheFile {
+	initOnce.Do(initCache)
+
+	return defaultCache
+}
