--- a/auth/hook.go
+++ b/auth/hook.go
@@ -123,7 +123,7 @@ func (a *HookAuth) GetValues(s string) {
 	s = strings.ReplaceAll(s, "\r\n", "\n")
 
 	// iterate input lines
-	for val := range strings.Lines(s) {
+	for _, val := range strings.Split(s, "\n") {
 		v := strings.SplitN(val, "=", 2)
 
 		// skips non key and value format
--- a/go.mod
+++ b/go.mod
@@ -1,6 +1,6 @@
 module github.com/filebrowser/filebrowser/v2
 
-go 1.24
+go 1.23.11
 
 require (
 	github.com/asdine/storm/v3 v3.2.1
